### PR TITLE
Display string enum pipe

### PR DESCRIPTION
**Pull request by jake-bassett**
## Description
We keep using the same code over and over. Turning it into a const method and pipe.

### CODE REVIEW

File: projects/common/src/utilities/formatters/enum/display-string-enum.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { displayStringEnum } from './display-string-enum';
+
+@Pipe({
+  name: 'htDisplayStringEnum'
+})
+export class DisplayStringEnumPipe implements PipeTransform {
```
Comment (anandtiwary) [inline projects/common/src/utilities/formatters/enum/display-string-enum.pipe.ts:7]: Why not use displayString?

Comment (jake-bassett) [reply projects/common/src/utilities/formatters/enum/display-string-enum.pipe.ts:7]: Hmm. Not sure that will work, or at least it shouldn't. The `displayString` method shouldn't replace `-` or `_`. It should just basically String(<arg>) any input but not format it... with some extra logic for null checks, arrays, etc.

Comment (aaron-steinfeld) [reply projects/common/src/utilities/formatters/enum/display-string-enum.pipe.ts:7]: I don't think we want to overload display string with this behavior (for example "my-value" would become "My Value", which we don't want the main string pipe doing)

**Review by aaron-steinfeld — COMMENTED** [review]
lgtm, tests please (at least on the const function, the pipe is trivial enough to not matter)

File: projects/common/src/utilities/formatters/enum/display-string-enum.ts [hunk 2]
```diff
@@ -5,5 +5,9 @@ export const displayStringEnum = (provided?: string): string => {
     return '-';
   }
 
-  return startCase(provided.toLowerCase());
+  // This removes dashes and underscores and gives all words initial caps
+  const startCased = startCase(provided);
+
+  // We only want the first word capitalized.
```
Comment (aaron-steinfeld) [inline projects/common/src/utilities/formatters/enum/display-string-enum.ts:11]: If that's what we want, what about `capitalize(lowerCase(provided))` - believe that works the same way as startCase:
```
_.lowerCase('--Foo-Bar--');
// => 'foo bar'
```

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/common/src/utilities/formatters/enum/display-string-enum.ts (hunk 2 above)
Comment (jake-bassett) [reply projects/common/src/utilities/formatters/enum/display-string-enum.ts:11]: That does look cleaner, but I think this one fails for camelCase and PascalCase.

Comment (aaron-steinfeld) [reply projects/common/src/utilities/formatters/enum/display-string-enum.ts:11]: image

Comment (anandtiwary) [reply projects/common/src/utilities/formatters/enum/display-string-enum.ts:11]: lodash should have ng pipes for each method. `'text' | lowercase | capitalize`

**Pull request merged**